sentry/src/main/java/io/sentry/protocol/User.java

```diff
@@ -57,6 +59,80 @@ public User(final @NotNull User user) {
     this.unknown = CollectionUtils.newConcurrentHashMap(user.unknown);
   }
 
+  /**
+   * Creates user from a map.
+   *
+   * @param map - The user data as map
+   * @return the user
+   */
+  @SuppressWarnings("unchecked")
+  public static User fromMap(@NotNull Map<String, Object> map, @NotNull SentryOptions options) {
+    final User user = new User();
+    Map<String, Object> unknown = null;
+
+    for (Map.Entry<String, Object> entry : map.entrySet()) {
+      Object value = entry.getValue();
+      switch (entry.getKey()) {
+        case JsonKeys.EMAIL:
+          user.email = (value instanceof String) ? (String) value : null;
+          break;
+        case JsonKeys.ID:
+          user.id = (value instanceof String) ? (String) value : null;
+          break;
+        case JsonKeys.USERNAME:
+          user.username = (value instanceof String) ? (String) value : null;
+          break;
+        case JsonKeys.SEGMENT:
+          user.segment = (value instanceof String) ? (String) value : null;
+          break;
+        case JsonKeys.IP_ADDRESS:
+          user.ipAddress = (value instanceof String) ? (String) value : null;
+          break;
+        case JsonKeys.DATA:
+          final Map<Object, Object> data =
+              (value instanceof Map) ? (Map<Object, Object>) value : null;
+          if (data != null) {
+            final ConcurrentHashMap<String, String> userData = new ConcurrentHashMap<>();
+            for (Map.Entry<Object, Object> dataEntry : data.entrySet()) {
+              if ((dataEntry.getKey() instanceof String)
+                  && (dataEntry.getValue() instanceof String)) {
```

Comment:
I know user's data is `<String,String>` for some reason, but what if lift up this instanceof check, and instead just call `dataEntry.getValue().toString()` if it's not null? Same for `other`.

I guess this way it's more flexible for the hybrid sdks to set whatever value in the data bag (except objects, but the object we have no idea how to handle anyway). We could also add more javadoc about that behavior.

Reply:
Updated the handling and added the missing fields. Would you add the doc directly to the new static methods?

Reply:
yeah, that should be enough

Reply:
Updated. If this is alright with you we should be ready to merge.